api_testing/apis/article_api.py

```diff
@@ -92,48 +62,46 @@ def create_article(self, **kwargs):
-            return response['message']
+    @api_factory('get')
+    def list_testplans_by_article_and_materialtype(self, materialtype_id=1, article_id=1, **kwargs):
+        api = '{}{}{}/{}'.format(self.url, self.END_POINTS['article_api']['list_testplans_by_article_and_materialtype'],
+                                 article_id, materialtype_id)
+        _payload = {}
+        return api, _payload, kwargs
 
-    def list_articles_by_materialtype(self, materialtype_id=1, name='', is_archived=0):
+    @api_factory('get')
+    def list_articles_by_materialtype(self, materialtype_id=1, name='', is_archived=0, **kwargs):
         api = '{}{}{}/{}?name={}'.format(self.url, self.END_POINTS['article_api']['list_articles_by_materialtype'],
                                          materialtype_id, is_archived, name)
-        self.info('GET : {}'.format(api))
-        response = self.session.get(api, params='', headers=self.headers, verify=False)
-        self.info('Status code: {}'.format(response.status_code))
-        data = response.json()
-        if data['status'] == 1:
-            return data['articles']
-        return []
+        _payload = {}
+        return api, _payload, kwargs
+
+
+class ArticleAPI(ArticleAPIFactory):
+    def delete_active_article(self, id=1):
+        response = self.archive_articles(ids=[str(id)])
+        if response['status'] == 1 and response['message'] == 'delete_success':
+            delete_response = self.delete_archived_article(id=id)
+            if delete_response['status'] == 1 and delete_response['message'] == 'hard_delete_success':
+                return True
+            else:
+                self.restore_articles(ids=[str(id)])
+                return False
+        else:
+            return False
 
     def get_active_articles_with_material_type(self):
         data = {}
-        articles = self.get_all_articles().json()['articles']
+        articles = self.get_all_articles()[0]['articles']
         for article in articles:
             material_type = article['materialType']
             if material_type not in data.keys():
                 data[material_type] = []
             data[material_type].append(article['name'])
         return data
 
-    def list_testplans_by_article_and_materialtype(self, materialtype_id=1, article_id=1):
-        api = '{}{}{}/{}'.format(self.url, self.END_POINTS['article_api']['list_testplans_by_article_and_materialtype'],
-                                 article_id, materialtype_id)
-        self.info('GET : {}'.format(api))
-        response = self.session.get(api, params='', headers=self.headers, verify=False)
-        self.info('Status code: {}'.format(response.status_code))
-        data = response.json()
-        if data['status'] == 1:
-            return data['testPlans']
-        return []
-
     def get_articles_with_no_testplans(self,**kwargs):
         response = self.get_all_articles(**kwargs)
         all_articles = response.json()['articles']
```

Comment:
shouldn't we update it with 
`        all_articles = response[0]['articles']
`

Reply:
done